src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -50,9 +50,19 @@ public override bool IsServerDelegatable(CallNode callNode, TexlBinding binding)
         {
             Contracts.AssertValue(callNode);
             Contracts.AssertValue(binding);
-            Contracts.Assert(binding.GetInfo(callNode).Function is UserDefinedFunction udf && udf.Binding != null);
-
-            return base.IsServerDelegatable(callNode, binding) || IsDelegatable;
+            Contracts.Assert(binding.GetInfo(callNode).Function is UserDefinedFunction udf && udf.Binding != null);
+
+            if (binding.DelegationHintProvider?.TryGetWarning(callNode, this, out var warning) ?? false)
+            {
+                SuggestDelegationHint(callNode, binding);
+            }
+
+            if (TryGetDataSource(callNode, binding, out var dataSource) && dataSource.IsDelegatable)
+            {
+                return true;
+            }

```

Comment:
Why do we need this? The binder doesn't mark the Top node as delegable?

Reply:
Binder doesn't mark Top as delegable. The setter `CheckAndMarkAsDelegatable(CallNode node)` in Binder ends up checking this method to see if marking as delegable, and it is false since setter and getter depend on the same field (the binary bit array).

Reply:
that's strange, we dont mark FirstNameNodes as delegable if they can be delegated? likely is the case , is it still possible to fix at UDF bind ?

Reply:
the problem is interface `IExternalDelegatableSymbol` doesn't implement on some data source types. Trying to fix and test in this way

Reply:
Please take a look at the new solution